src/pages/Admin/VenueList/VenueList.tsx

```diff
@@ -28,30 +30,34 @@ const VenueList: React.FC<VenueListProps> = ({
         </Link>
       </div>
       <ul className="page-container-adminsidebar-venueslist">
-        {relatedVenues.map((venue, index) => (
-          <li
-            key={index}
-            className={`${selectedVenueId === venue.id ? "selected" : ""} ${
-              canHaveSubvenues(venue) ? "camp" : ""
-            }`}
-          >
-            <Link to={`/admin-ng/venue/${venue.id}`}>{venue.name}</Link>
-            {isVenueWithRooms(venue) && venue.rooms && (
-              <ul className="page-container-adminsidebar-subvenueslist">
-                {venue.rooms.map((room, idx) => (
-                  <li
-                    key={idx}
-                    className={`${idx === roomIndex ? "selected" : ""}`}
-                  >
-                    <Link to={`/admin-ng/venue/${venue.id}?roomIndex=${idx}`}>
-                      {room.title}
-                    </Link>
-                  </li>
-                ))}
-              </ul>
-            )}
-          </li>
-        ))}
+        {isRelatedVenuesLoading ? (
+          <Loading />
+        ) : (
+          relatedVenues.map((venue, index) => (
+            <li
+              key={index}
+              className={`${selectedVenueId === venue.id ? "selected" : ""} ${
```

Comment:
This whole chunk should be extracted into a `useMemo`'d section, then should use our 'guard pattern' to return `<Loading />` when it's not yet ready.

Reply:
Got example of which the "guard pattern" is? I can't seem to connect the term with the appropriate refactor.

Reply:
Where we have a if (!foo) return type of like at the start of a block/function/etc that ‘shortcuts early’ or ‘guards’ the rest of the function/block. Followed by a single empty line of white space.

As opposed to nesting everything within it/etc.

Reply:
Oh, "fail fast", "early return", those terms I've used, and I've heard some one respond once with "we don't use that defensive programming".  As "guard pattern" on the other hand,  I thought of TS type guards only.

That aside, re reasoning was to allow some of the components to not be blocked by loading. like
```
  <Link to="/admin-ng/venue/creation" className="btn btn-primary">
    Create a venue
  </Link>
  ```
  so a memo block might also work.